flink-cdc-common/src/main/java/org/apache/flink/cdc/common/sink/MetadataApplier.java

```diff
@@ -19,13 +19,22 @@
 
 import org.apache.flink.cdc.common.annotation.PublicEvolving;
 import org.apache.flink.cdc.common.event.SchemaChangeEvent;
+import org.apache.flink.cdc.common.event.SchemaChangeEventType;
+import org.apache.flink.cdc.common.exceptions.SchemaEvolveException;
 
 import java.io.Serializable;
+import java.util.Set;
 
 /** {@code MetadataApplier} is used to apply metadata changes to external systems. */
 @PublicEvolving
 public interface MetadataApplier extends Serializable {
 
+    /** Checks if this metadata applier should handle this event type. */
+    boolean acceptsSchemaEvolutionType(SchemaChangeEventType schemaChangeEventType);
+
+    /** Checks what kind of schema change events downstream can handle. */
+    Set<SchemaChangeEventType> getSupportedSchemaEvolutionTypes();
+
```

Comment:
Please consider the compatibility when change existing public API. Imaging the case that user has a custom pipeline connector, is it still work after he/she bump the flink cdc version?

Reply:
I've tried to add reasonable `default` implementations for new methods to break less downstream code.